src/file_pipe_log/pipe_builder.rs

```diff
@@ -461,14 +460,28 @@ impl<F: FileSystem> DualPipesBuilder<F> {
                 while written < target_file_size {
                     writer.write_all(&buf).unwrap_or_else(|e| {
                         warn!("failed to build recycled file, err: {}", e);
+                        no_enough_space = true;
                     });
                     written += buf.len();
                 }
+                if no_enough_space {
```

Comment:
Let's play it safe, there's little benefit to have partially prefilled files anyway:
```
if no_space {
  target = 0;
}
```